### ISSUE TITLE
PANIC: called `Option::unwrap()` on a `None` value in query-engine/core/src/response_ir/internal.rs:364:77

### ISSUE THREAD

**Issue by arvind-plaxonic** [prisma/prisma#12126]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v17.4.0            | 
| OS              | debian-openssl-1.1.x|
| Prisma Client   | 3.10.0             |
| Query Engine    | 0.1.0              |
| Database        | sqlite             |

## Logs
```
  prisma:tryLoadEnv Environment variables not found at null  
  prisma:tryLoadEnv Environment variables not found at undefined  
  prisma:tryLoadEnv No Environment variables loaded  
  prisma:tryLoadEnv Environment variables not found at null  
  prisma:tryLoadEnv Environment variables not found at undefined  
  prisma:tryLoadEnv No Environment variables loaded  
  prisma:client clientVersion: 3.10.0  
  prisma:client clientEngineType: library  
  prisma:client:libraryEngine internalSetup  
  prisma:client:libraryEngine Searching for Query Engine Library in /home/arvind/cms/node_modules/.prisma/client  
  prisma:client:libraryEngine loadEngine using /home/arvind/cms/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node  
  prisma:client:libraryEngine library starting  
  prisma:client:libraryEngine library started  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +9s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +6s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
```

## Client Snippet
```ts
// PLEASE FILL YOUR CODE SNIPPET HERE
```

## Schema
```prisma
// This file is automatically generated by Keystone, do not modify it manually.
// Modify your Keystone config when you want to change this.

datasource sqlite {
  url      = env("DATABASE_URL")
  provider = "sqlite"
}

generator client {
  provider = "prisma-client-js"
  output   = "node_modules/.prisma/client"
}

model User {
  id       String @id @default(cuid())
  name     String @default("")
  email    String @unique @default("")
  password String
  posts    Post[] @relation("Post_author")
}

model NavBarItem {
  id           String       @id @default(cuid())
  label        String       @default("")
  link         String       @default("")
  children     NavBarItem[] @relation("NavBarItem_children")
  parent       NavBarItem[] @relation("NavBarItem_children")
  page         Page[]       @relation("NavBarItem_page")
  CallToAction String?      @default("menu")
}

model Page {
  id                String           @id @default(cuid())
  title             String           @default("")
  slug              String           @default("")
  description       String           @default("")
  sections          Section[]        @relation("Page_sections")
  FooterPartPosts   FooterPartPost[] @relation("FooterPartPost_page")
  NavBarItems       NavBarItem[]     @relation("NavBarItem_page")
  Posts             Post[]           @relation("Page_Posts")
  tags              Tag?             @relation("Page_tags", fields: [tagsId], references: [id])
  tagsId            String?          @map("tags")
  from_Article_Page Article[]        @relation("Article_Page")

  @@index([slug])
  @@index([tagsId])
}

model Section {
  id               String    @id @default(cuid())
  title            String    @default("")
  description      String    @default("")
  Content          String    @default("[{\"type\":\"paragraph\",\"children\":[{\"text\":\"\"}]}]")
  ButtonName       String    @default("")
  Url              String    @default("")
  FeactureImageUrl String    @default("")
  OtherImageUrl    String    @default("")
  order            Int?
  Type             String?   @default("article")
  page             Page[]    @relation("Page_sections")
  articles         Article[] @relation("Article_section")
  data             String    @default("")
}

model Article {
  id               String    @id @default(cuid())
  title            String    @default("")
  SubTitle         String    @default("")
  ExtraText        String    @default("")
  tags             Tag?      @relation("Article_tags", fields: [tagsId], references: [id])
  tagsId           String?   @map("tags")
  Page             Page[]    @relation("Article_Page")
  Content          String    @default("[{\"type\":\"paragraph\",\"children\":[{\"text\":\"\"}]}]")
  FeactureImageUrl String    @default("")
  MediaFile        String    @default("")
  OtherImageUrl    String    @default("")
  ButtonName       String    @default("")
  Url              String    @default("")
  order            Int?
  section          Section[] @relation("Article_section")

  @@index([tagsId])
}

model Post {
  id           String    @id @default(cuid())
  title        String    @default("")
  category     String    @default("")
  creationDate DateTime?
  imageUrl     String    @default("")
  Content      String    @default("[{\"type\":\"paragraph\",\"children\":[{\"text\":\"\"}]}]")
  author       User?     @relation("Post_author", fields: [authorId], references: [id])
  authorId     String?   @map("author")
  tags         Tag[]     @relation("Post_tags")
  page         Page[]    @relation("Page_Posts")

  @@index([authorId])
}

model FooterPartPost {
  id        String  @id @default(cuid())
  title     String  @default("")
  PageLink  String  @default("")
  ExtraText String  @default("")
  order     Int?
  tag       Tag?    @relation("FooterPartPost_tag", fields: [tagId], references: [id])
  tagId     String? @map("tag")
  page      Page[]  @relation("FooterPartPost_page")

  @@index([tagId])
}

model Tag {
  id                      String           @id @default(cuid())
  name                    String           @default("")
  from_Page_tags          Page[]           @relation("Page_tags")
  from_Article_tags       Article[]        @relation("Article_tags")
  from_Post_tags          Post[]           @relation("Post_tags")
  from_FooterPartPost_tag FooterPartPost[] @relation("FooterPartPost_tag")
}
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by janpio** [comment]
Can you plese provide some more information what kind of query you were executing when this happened?

**Comment by arvind-plaxonic** [comment]
Hello @janpio Sir,

Actually I have create CMS in keystonejs, When i am running dashboard, it is showing:  please see:

An error occurred handling a request for the Admin UI: Error: Prisma error
    at Object.prismaError (/home/arvind/cms/node_modules/@keystone-6/core/dist/graphql-errors-4f230fdf.cjs.dev.js:27:12)
    at runWithPrisma (/home/arvind/cms/node_modules/@keystone-6/core/dist/types-for-lists-c9e0cd07.cjs.dev.js:30:25)
    at Object.count (/home/arvind/cms/node_modules/@keystone-6/core/dist/types-for-lists-c9e0cd07.cjs.dev.js:911:17)
    at async Promise.all (index 0)
    at Object.runGraphQL [as run] (/home/arvind/cms/node_modules/@keystone-6/core/dist/initConfig-2fd803cd.cjs.dev.js:2817:20)
    at Object.count (/home/arvind/cms/node_modules/@keystone-6/core/dist/initConfig-2fd803cd.cjs.dev.js:2496:24)
    at pageMiddleware (/home/arvind/cms/node_modules/@keystone-6/auth/dist/keystone-6-auth.cjs.dev.js:894:21)
    at Object.pageMiddleware (/home/arvind/cms/node_modules/@keystone-6/auth/dist/keystone-6-auth.cjs.dev.js:1105:43)
    at /home/arvind/cms/node_modules/@keystone-6/core/dist/createAdminUIMiddleware-abacc995.cjs.dev.js:178:30 {
  locations: [ { line: 1, column: 35 } ],
  path: [ 'count' ],
  extensions: {
    debug: {
      message: '\n' +
        'Invalid `prisma.user.count()` invocation:\n' +
        '\n' +
        '\n' +
        '  PANIC: called `Option::unwrap()` on a `None` value in query-engine/core/src/response_ir/internal.rs:364:77\n' +
        '\n' +
        'This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.\n' +
        '\n' +
        'https://github.com/prisma/prisma/issues/new?body=Hi+Prisma+Team%21+My+Prisma+Client+just+crashed.+This+is+the+report%3A%0A%23%23+Versions%0A%0A%7C+Name++++++++++++%7C+Version++++++++++++%7C%0A%7C-----------------%7C--------------------%7C%0A%7C+Node++++++++++++%7C+v17.4.0++++++++++++%7C+%0A%7C+OS++++++++++++++%7C+debian-openssl-1.1.x%7C%0A%7C+Prisma+Client+++%7C+3.10.0+++++++++++++%7C%0A%7C+Query+Engine++++%7C+0.1.0++++++++++++++%7C%0A%7C+Database++++++++%7C+sqlite+++++++++++++%7C%0A%0A%0A%0A%23%23+Logs%0A%60%60%60%0A++prisma%3AtryLoadEnv+Environment+variables+not+found+at+null++%0A++prisma%3AtryLoadEnv+Environment+variables+not+found+at+undefined++%0A++prisma%3AtryLoadEnv+No+Environment+variables+loaded++%0A++prisma%3AtryLoadEnv+Environment+variables+not+found+at+null++%0A++prisma%3AtryLoadEnv+Environment+variables+not+found+at+undefined++%0A++prisma%3AtryLoadEnv+No+Environment+variables+loaded++%0A++prisma%3Aclient+clientVersion%3A+3.10.0++%0A++prisma%3Aclient+clientEngineType%3A+library++%0A++prisma%3Aclient%3AlibraryEngine+internalSetup++%0A++prisma%3Aclient%3AlibraryEngine+Searching+for+Query+Engine+Library+in+%2Fhome%2Farvind%2Fcms%2Fnode_modules%2F.prisma%2Fclient++%0A++prisma%3Aclient%3AlibraryEngine+loadEngine+using+%2Fhome%2Farvind%2Fcms%2Fnode_modules%2F.prisma%2Fclient%2Flibquery_engine-debian-openssl-1.1.x.so.node++%0A++prisma%3Aclient%3AlibraryEngine+library+starting++%0A++prisma%3Aclient%3AlibraryEngine+library+started++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B5s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B3s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B2s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B22s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B2s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B2s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%2B12s%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A++prisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true++%0A%60%60%60%0A%0A%23%23+Client+Snippet%0A%60%60%60ts%0A%2F%2F+PLEASE+FILL+YOUR+CODE+SNIPPET+HERE%0A%60%60%60%0A%0A%23%23+Schema%0A%60%60%60prisma%0A%2F%2F+PLEASE+ADD+YOUR+SCHEMA+HERE+IF+POSSIBLE%0A%60%60%60%0A%0A%23%23+Prisma+Engine+Query%0A%60%60%60%0A%7B%22X%22%3A%7B%7D%7D%0A%60%60%60%0A&title=PANIC%3A+called+%60Option%3A%3Aunwrap%28%29%60+on+a+%60None%60+value+in+query-engine%2Fcore%2Fsrc%2Fresponse_ir%2Finternal.rs%3A364%3A77&template=bug_report.md\n' +
        '\n' +
        'If you want the Prisma team to look into it, please open the link above 🙏\n' +
        'To increase the chance of success, please post your schema and a snippet of\n' +
        'how you used Prisma Client in the issue. \n'
    },
    code: 'KS_PRISMA_ERROR'
  }
}
Sir pls help me

**Comment by janpio** [comment]
Can you enable Prisma logging? That should log the query that is executed there exactly: https://pris.ly/d/logging + https://pris.ly/d/debugging

**Comment by ivosabev** [comment]
I am experiencing the same issue with `3.10.0`, it is not present with `3.9.2` and with `3.11.0-dev.52`

**Comment by janpio** [comment]
Same questions to you then @ivosabev - see above. We need more information to be able to do anything.

**Comment by erikbjoern** [comment]
I have the same problem with `3.10.0`, not present with `3.9.2`

It's just a matter of querying one of our models - only this one, and it does not matter if I use `findUnique` or `findFirst`.  I tried commenting out all relations and then it worked, but when I tried to uncomment them one by one the behaviour wasn't consistent, I couldn't figure out if any specific relation was the problem.

## Versions:
| Name                  | Version             |
|-----------------|----------------|
| Node                   | v16.14.0            | 
| OS                       | darwin              |
| Prisma Client      | 3.10.0               |
| Query Engine     | 0.1.0                  |
| Database            | mysql                |

## Rust backtrace:
```
   0:        0x11d3bdc31 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h90c059b532d34bdd
   1:        0x11d3dffcb - core::fmt::write::hefb311138778afb5
   2:        0x11d3b6d5a - std::io::Write::write_fmt::h0e0ed87b18ae870c
   3:        0x11d3bf9b5 - std::panicking::default_hook::{{closure}}::h43f0e51e5825bd5b
   4:        0x11d3bf59f - std::panicking::default_hook::ha1c3473cbcf391f7
   5:        0x11c08c08d - query_engine::engine::set_panic_hook::{{closure}}::h5fd99e2c8cab0680
   6:        0x11d3c00c6 - std::panicking::rust_panic_with_hook::h0db3c4ba5fb4dc12
   7:        0x11d3bfb14 - std::panicking::begin_panic_handler::{{closure}}::hc6d8c92efeef5a19
   8:        0x11d3be0a7 - std::sys_common::backtrace::__rust_end_short_backtrace::h7ffb59e898e02a83
   9:        0x11d3bfaaa - _rust_begin_unwind
  10:        0x11d44862f - core::panicking::panic_fmt::h7d73f67464d78916
  11:        0x11d448587 - core::panicking::panic::h9721d5a8c5d33ad7
  12:        0x11c2a25d2 - query_core::response_ir::internal::serialize_objects::hcbb3e5a35bf1b163
  13:        0x11c29e881 - query_core::response_ir::internal::serialize_record_selection::hff326efe6929dc94
  14:        0x11c29b07b - query_core::response_ir::internal::serialize_internal::hd01e18f6cb103f47
  15:        0x11c330798 - query_core::response_ir::ir_serializer::IrSerializer::serialize::h63503a890a90ba56
  16:        0x11c04358d - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hfc544138efa64996
  17:        0x11c0340c5 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hbef1d4078c8baba1
  18:        0x11c01a9e5 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h3320e01c5126c526
  19:        0x11c03c463 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::he6a8715be671cc2c
  20:        0x11c016c81 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h1f64163b60fc3f60
  21:        0x11c020d7f - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h5493be79bdc17417
  22:        0x11c0266b0 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7a0ac9d980b9ee53
  23:        0x11c03e6ba - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hedcd61ded9f36990
  24:        0x11c027f7a - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7f76a67c327697d5
  25:        0x11bfcb333 - tokio::runtime::task::harness::Harness<T,S>::poll::hfe010040e1d19aaa
  26:        0x11d14de4d - std::thread::local::LocalKey<T>::with::h21a4b8b337b67c27
  27:        0x11d148e8f - tokio::runtime::thread_pool::worker::Context::run_task::h59a149308d0766fd
  28:        0x11d148499 - tokio::runtime::thread_pool::worker::Context::run::hc868bc7b234ff7f5
  29:        0x11d135825 - tokio::macros::scoped_tls::ScopedKey<T>::set::h6e800fa1ea63250e
  30:        0x11d147964 - tokio::runtime::thread_pool::worker::run::h5e69089734701f92
  31:        0x11d14bca9 - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::h4ec0494860eecc5d
  32:        0x11d13fc13 - tokio::runtime::task::harness::Harness<T,S>::poll::h5bb4edf7863a87fc
  33:        0x11d12bac7 - tokio::runtime::blocking::pool::Inner::run::h6a23c4abad6b4fef
  34:        0x11d133cfe - std::sys_common::backtrace::__rust_begin_short_backtrace::ha216731a1b94a7c1
  35:        0x11d14e52c - core::ops::function::FnOnce::call_once{{vtable.shim}}::h3a881747160f83a4
  36:        0x11d3c5a77 - std::sys::unix::thread::Thread::new::thread_start::h13359fe05abc98e9
  37:     0x7ff814e404f4 - __pthread_start
  ```

**Comment by janpio** [comment]
Thanks for the report @erikbjoern. Can you share the schema of the model(s)? Does it depend on the data in the database table or does it also happen on an empty, newly created database of that schema?

Also: https://github.com/prisma/prisma/issues/12126#issuecomment-1057971305 If you can report the actual query (both Prisma and SQL) that triggers this would be most helpful.

**Comment by erikbjoern** [comment]
@Jolg42 We are no longer experiencing the issue. I'm not sure which version resolved it, but at least since `3.12.0` it's been working. We did not perform any `count()` in the failing query - as I mentioned above it was only a single model that behaved this way, even if we didn't pass any params at all, just a simple `prisma.model.findFirst()`.

I could post the schema but it is rather gigantic with many interconnected relations

**Comment by janpio** [comment]
@arvind-plaxonic @ivosabev Are you still experiencing this problem? If so, can you please provide some additional information that will help us reproduce this so we can actually fix it? Thanks. Otherwise we will have to close this issue.

**Comment by Jolg42** [comment]
This problem looks like a duplicate of https://github.com/prisma/prisma/issues/12105 which has a comment that it was fixed in `3.11.0` https://github.com/prisma/prisma-engines/pull/2732

@arvind-plaxonic @ivosabev Could you try `3.11.0` or a more recent version and confirm it was fixed?

**Comment by janpio** [comment]
Closing then until someone provides more information. Please comment or open a new issue. Thanks!
